substrate/client/rpc-spec-v2/src/chain_head/chain_head.rs

```diff
@@ -296,19 +329,33 @@ where
 			Err(_) => return Err(ChainHeadRpcError::InvalidBlock.into()),
 		};
 
-		self.client
-			.header(hash)
-			.map(|opt_header| opt_header.map(|h| hex_string(&h.encode())))
-			.map_err(|err| ChainHeadRpcError::InternalError(err.to_string()))
+		let client = self.client.clone();
+		let result = spawn_blocking(&self.executor, async move {
+			let _block_guard = block_guard;
+
+			client
+				.header(hash)
+				.map(|opt_header| opt_header.map(|h| hex_string(&h.encode())))
+				.map_err(|err| ChainHeadRpcError::InternalError(err.to_string()))
+		});
+		result.await.unwrap_or_else(|_| Ok(None))
 	}
 
-	fn chain_head_unstable_storage(
+	async fn chain_head_unstable_storage(
 		&self,
+		connection_details: ConnectionDetails,
 		follow_subscription: String,
 		hash: Block::Hash,
 		items: Vec<StorageQuery<String>>,
 		child_trie: Option<String>,
 	) -> ResponsePayload<'static, MethodResponse> {
+		if !self
+			.rpc_connections
+			.contains_token(connection_details.id(), &follow_subscription)
+		{
+			return ResponsePayload::success(MethodResponse::LimitReached);
```

Comment:
> This return value indicates the call couldn't be started because the server is overloaded, or that the followSubscription is invalid or stale.

I think it's quite strange to use `LimitReached` here but according to the spec...